wnfs/src/lib.rs

```diff
@@ -8,26 +8,27 @@
 //! It exposes an immutable API, extending WNFS immutable nature to the in-memory representation of the file system.
 #![deny(unsafe_code)]
 
-mod common;
+mod error;
 pub mod private;
 pub mod public;
 mod traits;
-
-pub use common::*;
-pub use private::{
-    hamt::{Hasher, Pair},
-    namefilter::Namefilter,
-    PrivateDirectory, PrivateFile, PrivateNode, PrivateOpResult,
-};
-pub use public::{PublicDirectory, PublicFile, PublicNode, PublicOpResult};
-pub use traits::*;
-
+mod utils;
 //--------------------------------------------------------------------------------------------------
 // Re-exports
 //--------------------------------------------------------------------------------------------------
-
 pub use libipld;
-
+pub use private::{PrivateDirectory, PrivateFile, PrivateNode, PrivateOpResult};
+pub use public::{PublicDirectory, PublicFile, PublicNode, PublicOpResult};
+pub use traits::*;
 pub mod rand_core {
     pub use rand_core::RngCore;
 }
+pub mod common {
+    pub use wnfs_common::*;
+}
+pub mod hamt {
+    pub use wnfs_hamt::*;
+}
+pub mod namefilter {
+    pub use wnfs_namefilter::*;
+}
```

Comment:
Just so I understand this correctly:
The reasoning for re-exporting generally is that users of `wnfs` shouldn't have to import another library if they want to make use of its public API, right?
So, for `common` and `namefilter` that makes sense to me. But do `hamt` details actually appear in the public API or is everything hidden behind the `PrivateForest`?
Can we just keep the HAMT detail hidden & not reexport?

Reply:
There is the `Hasher` trait

Reply:
Right.

What if we put that into wnfs-common?